### PR TITLE
feat: support cpu provision of aqc

### PR DESCRIPTION
**Pull request by cheney-lin**
#### What type of PR is this?

Features

### CODE REVIEW

File: pkg/apis/config/v1alpha1/adminqos.go [hunk 1]
```diff
@@ -131,6 +131,45 @@ type CPUAdvisorConfig struct {
...
+	CPUProvisionConfig *CPUProvisionConfig `json:"cpuProvisionConfig"`
+}
+
+type IndicatorTargetConfiguration struct {
+	Name   string  `json:"name"`
+	Target float64 `json:"target"`
+}
+
+// FirstOrderPIDParams holds parameters for pid controller in rama policy
+type FirstOrderPIDParams struct {
```
Comment (luomingmeng) [inline pkg/apis/config/v1alpha1/adminqos.go:145]: Why add prefix "FirstOrder" to PIDParameters?

File: pkg/apis/config/v1alpha1/adminqos.go [hunk 2]
```diff
@@ -131,6 +131,45 @@ type CPUAdvisorConfig struct {
...
+	DeadbandLowerPct     float64 `json:"deadbandLowerPct"`
+}
+
+type PolicyRamaConfiguration struct {
+	PIDParameters                   map[string]FirstOrderPIDParams `json:"PIDParameters,omitempty"`
+	EnableBorwein                   bool                           `json:"enableBorwein,omitempty"`
+	EnableBorweinModelResultFetcher bool                           `json:"enableBorweinModelResultFetcher,omitempty"`
+}
+
+type ControlKnobConstraints struct {
```
Comment (luomingmeng) [inline pkg/apis/config/v1alpha1/adminqos.go:162]: why not use list instead of map?

File: pkg/apis/config/v1alpha1/adminqos.go [hunk 3]
```diff
@@ -131,6 +131,45 @@ type CPUAdvisorConfig struct {
...
+	Kdp                  float64 `json:"kdp"`
+	Kdn                  float64 `json:"kdn"`
+	AdjustmentUpperBound float64 `json:"adjustmentUpperBound"`
+	AdjustmentLowerBound float64 `json:"adjustmentLowerBound"`
+	DeadbandUpperPct     float64 `json:"deadbandUpperPct"`
+	DeadbandLowerPct     float64 `json:"deadbandLowerPct"`
+}
+
+type PolicyRamaConfiguration struct {
+	PIDParameters                   map[string]FirstOrderPIDParams `json:"PIDParameters,omitempty"`
```
Comment (luomingmeng) [inline pkg/apis/config/v1alpha1/adminqos.go:157]: Using list instead of map is better

File: pkg/apis/config/v1alpha1/adminqos.go [hunk 4]
```diff
@@ -131,6 +131,45 @@ type CPUAdvisorConfig struct {
...
+
+type ControlKnobConstraints struct {
+	RestrictControlKnobMaxUpperGap      map[string]float64 `json:"restrictControlKnobMaxUpperGap,omitempty"`
+	RestrictControlKnobMaxLowerGap      map[string]float64 `json:"restrictControlKnobMaxLowerGap,omitempty"`
+	RestrictControlKnobMaxUpperGapRatio map[string]float64 `json:"restrictControlKnobMaxUpperGapRatio,omitempty"`
+	RestrictControlKnobMaxLowerGapRatio map[string]float64 `json:"restrictControlKnobMaxLowerGapRatio,omitempty"`
+}
+
+type CPUProvisionConfig struct {
+	IndicatorTargets       map[string][]IndicatorTargetConfiguration `json:"indicatorTargets,omitempty"`
```
Comment (luomingmeng) [inline pkg/apis/config/v1alpha1/adminqos.go:170]: the same as before and it is better use list instead of map and add more comments

**Review by luomingmeng — APPROVED** [review]

**Review by nightmeng — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**